include/swift/Runtime/Debug.h

```diff
@@ -134,6 +134,56 @@ void dumpStackTraceEntry(unsigned index, void *framePC,
 LLVM_ATTRIBUTE_NOINLINE
 void printCurrentBacktrace(unsigned framesToSkip = 1);
 
+/// Debugger breakpoint ABI. This structure is passed to the debugger (and needs
+/// to be stable) and describes extra information about a fatal error or a
+/// non-fatal warning, which should be logged as a runtime issue. Please keep
+/// all integer values pointer-sized.
+struct RuntimeErrorDetails {
+  // ABI version, needs to be "1" currently.
+  uintptr_t version;
+
+  // A short hyphenated string describing the type of the issue, e.g.
+  // "precondition-failed" or "exclusivity-violation".
+  const char *errorType;
+
+  // Description of the current thread's stack position.
+  const char *currentStackDescription;
+
+  // Address of some associated object (if there's any).
+  void *memoryAddress;
+
+  // A structure describing an extra thread (and its stack) that is related.
+  struct Thread {
+    const char *description;
+    uint64_t threadID;
+    uintptr_t numFrames;
+    void **frames;
+  };
+
+  // Number of extra threads (excluding the current thread) that are related,
+  // and the pointer to the array of extra threads.
+  uintptr_t numExtraThreads;
+  Thread *threads;
+
+  // Description of a related object (other than 'memoryAddress').
+  struct RelatedObject {
+    const char *description;
+    const char *filename;
+    uintptr_t line;
+    uintptr_t column;
```

Comment:
This is meant for SE-0160, and potentially for future uses as "notes" for runtime issues.  I found out that the objc thunks don't have debug info, so the debugger won't point to the declaration on its own.

Any suggestions?

We could also drop these "related objects" completely, and just provide a stacktrace.  Users would then have to figure out where the declaration is themselves by looking at stderr.